sky/utils/controller_utils.py

```diff
@@ -93,17 +83,19 @@ def check_cluster_name_hint(self) -> str:
             cluster_name=self.cluster_name)
 
 
+# TODO: refactor controller class to not be an enum.
 class Controllers(enum.Enum):
     """Skypilot controllers."""
     # NOTE(dev): Keep this align with
     # sky/cli.py::_CONTROLLER_TO_HINT_OR_RAISE
     JOBS_CONTROLLER = _ControllerSpec(
         controller_type='jobs',
         name='managed jobs controller',
-        candidate_cluster_names=[
-            managed_job_utils.JOB_CONTROLLER_NAME,
-            managed_job_utils.LEGACY_JOB_CONTROLLER_NAME
```

Comment:
I guess we will lose support for legacy controller name - is that right?

Reply:
Yes, we have to give up some backward compatibility for the simplicity of the implementation. Since we have 0.8 merged before this PR with backward compatibility, it should be fine to remove the backward compatibility here.